### DISCUSSION TITLE
Lighting bug in underground areas

### DISCUSSION THREAD

**Discussion by MrSeyker** [OGSR/OGSR-Engine#306] [Q&A]
I noticed that in OGSR areas bellow the ground, such as Sidorovich's bunker, would be cast with light from the sun/flares whenever the weather system positions them under the horizon.

I've been told the same bug can be experienced in the Bar, although I've not reached there yet to confirm.

I've partially remedied the issue by making sure the sun_position in the vanilla weather files isn't set higher than -16  for the first value before the the engine is told to turn off the sun via sun_color.

This change in the lighting system was addressed in CS/COP by surrounding underground elements with a material that disables sun illumination, but this requires setting them up in the level editor.

The older SOC assets have no such bounding boxes (I suppose the objects themselves would occlude the sun?).

This results in lighting bugs in just about every single weather mod out there.

Can the lighting behavior of the SoC engine be replicated?

**Comment by xrSimpAI** [comment]
The problem is that in SoC, the sunlight is not coming from the sun. The sun can be in one place, and the sunlight / shadows can be completely on the other side. In the ogsr, the sunlight comes directly from the sun, as in the CoP. Therefore, location bugs become noticeable, the absence of double-sided polygons for objects through which the sun shines through.

**Reply by MrSeyker** [reply]
I don't think that's true, I remember I had a weather mod for SoC that would position the sun to always rise in the east side of the map and the shadows of the trees and houses would always correspond the position of the sun flare in the sky.

That's in fact why I ended up removing the mod, because while it looked great in dynamic lighting, the moment In switched to static lighting object shadows would always end up facing towards the sun flare.

**Reply by xrSimpAI** [reply]
Well, by the way, if you turn on `renderer renderer_r2a`, there will be a static sun, as in SoC. Do you need this, do I understand correctly?

**Reply by MrSeyker** [reply]
I've tested all the OGSR renderers. None can get rid of the sun shining underground (unless r2a is only available through console and not a menu option).

Like I said, I came up with a "fix" in the weather configs, but that changes the mood of the lighting (I can't get the low-on-horizon sunrise and sunsets of vanilla anymore due to this).

Can't the engine be updated so that light can't pass through one sided polygons? I mean, from the side of the sun there's nothing to obstruct the light it's casting, but from the other side a wall is being rendered.

**Reply by xrSimpAI** [reply]
> Can't the engine be updated so that light can't pass through one sided polygons? I mean, from the side of the sun there's nothing to obstruct the light it's casting, but from the other side a wall is being rendered.

Unfortunately, I have not seen such methods.
